sdks/java/io/file-based-io-tests/src/test/java/org/apache/beam/sdk/io/text/TextIOIT.java

```diff
@@ -81,13 +81,38 @@ public static void setup() throws ParseException {
         .as(IOTestPipelineOptions.class);
 
     numberOfTextLines = options.getNumberOfRecords();
-    filenamePrefix = appendTimestamp(options.getFilenamePrefix());
+    filenamePrefix = resolveProtocolAndPath(options);
   }
 
   private static String appendTimestamp(String filenamePrefix) {
     return String.format("%s_%s", filenamePrefix, new Date().getTime());
   }
 
+  private static String resolveProtocolAndPath(IOTestPipelineOptions options) {
```

Comment:
I'm not sure why we need to parse and reassemble protocol here. We shouldn't have to do this if we ask user to give the full prefix that includes the protocol.

Reply:
Sure. However if in the future we will have kind of "validator" that will validate test input parameters, then it would reassemble `--filenamePrefix` I guess.

Reply:
Yeah, we can have validate that makes sure that fileSystem property and fileNamePrefix match.